File: solutions/c/space-age/1/space_age.c
#include "space_age.h"
float age(planet_t planet, int64_t seconds) {
    float res = seconds / 31557600.0;
    switch(planet) {
        case 0:
            res /= 0.2408467;
            break;
        case 1:
            res /= 0.61519726;
            break;
        case 2:
            break;
        case 3:
            res /= 1.8808158;
            break;
        case 4:
            res /= 11.862615;
            break;
        case 5:
            res /= 29.447498;
            break;
        case 6:
            res /= 84.016846;
            break;
        case 7:
            res /= 164.79132;
            break;
        default:
            res = -1;
    }
    return res;
}

File: solutions/go/space-age/1/space_age.go
package space

type Planet string

func Age(seconds float64, planet Planet) float64 {
    data := make(map[Planet]float64)
    data["Mercury"] = seconds / (31557600.0 * 0.2408467)
    data["Venus"] = seconds / (31557600.0 * 0.61519726)
    data["Earth"] = seconds / 31557600.0
    data["Mars"] = seconds / (31557600.0 * 1.8808158)
    data["Jupiter"] = seconds / (31557600.0 * 11.862615)
    data["Saturn"] = seconds / (31557600.0 * 29.447498)
    data["Uranus"] = seconds / (31557600.0 * 84.016846)
    data["Neptune"] = seconds / (31557600.0 * 164.79132)
    _, exists := data[planet]
    if exists {
		return data[planet]
    }
    return -1
}
